File: src/NzbDrone.Core/Books/Book.cs
using System;
using NzbDrone.Core.MediaItems;
using NzbDrone.Core.MediaTypes;

namespace NzbDrone.Core.Books
{
    public class Book : MediaItem
    {
        public Book()
        {
            MediaType = MediaType.Book;
        }

        public string Title { get; set; }
        public string SortTitle { get; set; }
        public string Description { get; set; }
        public string ForeignBookId { get; set; }
        public string Isbn { get; set; }
        public string Isbn13 { get; set; }
        public string Asin { get; set; }
        public int? PageCount { get; set; }
        public DateTime? ReleaseDate { get; set; }
        public string Publisher { get; set; }
        public string Language { get; set; }

        public int? SeriesPosition { get; set; }

        public override string GetTitle() => Title;
        public override int GetYear() => ReleaseDate?.Year ?? 0;

        public override string ToString()
        {
            return $"{Title} ({ReleaseDate?.Year})";
        }
    }
}

File: frontend/src/Book/Book.ts
import ModelBase from 'App/ModelBase';

interface Book extends ModelBase {
  title: string;
  sortTitle: string;
  description: string;
  foreignBookId: string;
  isbn: string;
  isbn13: string;
  asin: string;
  pageCount: number;
  releaseDate: string;
  publisher: string;
  language: string;
  monitored: boolean;
  qualityProfileId: number;
  path: string;
  rootFolderPath: string;
  added: string;
  tags: number[];
  lastSearchTime?: string;
  authorId?: number;
  bookSeriesId?: number;
  seriesPosition?: number;
  isSaving?: boolean;
}

export default Book;
